src/core/createCookie.js

```diff
@@ -9,125 +9,53 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
-    return cookieProxy;
-  }
-
-  const cookieName = `${ALLOY_COOKIE_NAME}_${propertyId}`;
-  let deserializedCookie;
-  let cookieHasBeenRead = false;
-
-  cookieProxy = {
-    get() {
-      // We don't read the cookie right when the cookie proxy is created
-      // because we don't know if the user has opted in. If the user
-      // hasn't opted in, we're legally obligated to not read the cookie.
-      // If a component tries to read something off the cookie though,
-      // we assume that the component has received word that the user
-      // has opted-in. The responsibility is on the component.
-      if (cookieHasBeenRead) {
-        return deserializedCookie;
-      }
-
-      const serializedCookie = cookie.get(cookieName);
-      deserializedCookie =
-        serializedCookie && safeJSONParse(serializedCookie, cookieName);
-      cookieHasBeenRead = true;
-      return deserializedCookie;
-    },
-    set(updatedCookie) {
-      deserializedCookie = updatedCookie;
-      cookie.set(cookieName, updatedCookie, {
-        expires: ALLOY_COOKIE_TTL,
-        domain: cookieDomain || getTopLevelCookieDomain(window, cookie)
-      });
-    }
-  };
-
-  cookieProxyByPropertyId[propertyId] = cookieProxy;
-
-  return cookieProxy;
-};
-
-// TODO: Support passing a configurable expiry in the config when creating this cookie.
-const createCookie = (componentNamespace, propertyId, cookieDomain = "") => {
-  const cookieProxy = createCookieProxy(propertyId, cookieDomain);
 
+export default (cookieProxy, componentNamespace) => {
   return {
     /**
-     * Returns the value from AlloyCookie for a prefix and a key.
-     * @param {...*} arg String key stored in alloy cookie under a component prefix.
+     * Returns a value from the Alloy cookie for a given key.
+     * @param {string} key
      */
-    get(name) {
+    get(key) {
       const currentCookie = cookieProxy.get();
       return (
         currentCookie &&
         currentCookie[componentNamespace] &&
-        currentCookie[componentNamespace][name]
+        currentCookie[componentNamespace][key]
       );
     },
     /**
-     * Updates the value  of a key from AlloyCookie for a prefix
-     * @param {...*} arg Strings with key and value to be stored in alloy cookie.
+     * Sets a value in the Alloy cookie for a given key.
+     * @param {string} key
+     * @param {string} value
      */
     set(key, value) {
       const currentCookie = cookieProxy.get() || {};
-      // Yes, we're mutating the object returned from the cookie proxy, but
-      // it's reasonably controlled since the side effects are all contained
-      // within this file.
-      currentCookie[componentNamespace] =
-        currentCookie[componentNamespace] || {};
-      currentCookie[componentNamespace][key] = value;
-      cookieProxy.set(currentCookie);
+      const updatedCookie = {
+        ...currentCookie,
+        [componentNamespace]: {
+          ...currentCookie[componentNamespace],
```

Comment:
Will this pass in IE 11?

Reply:
Yes. Babel's object spreading output doesn't try to use symbols or Array.from in IE.